src/App.js

```diff
@@ -3,15 +3,57 @@ import Header from './Header.js';
 import Footer from './Footer.js';
 import Main from './Main.js';
 import data from './data.json'
+import Modal from 'react-bootstrap/Modal'
 
 import './App.css';
 
 class App extends React.Component{
+  constructor(props){
+    super(props);
+    this.state={
+      show: false,
+      animal:{title: 'some beast'},
+      // title: this.props.animal.title
+    }
+  }
+
+  handleShow = (animal) => {
+    this.setState({
+      show: true,
+      animal
+    })
+  }
+  handleClose = () => {
+    this.setState ({
+      show:false
+    })
+  }
+  
   render(){
     return(
       <>
         <Header />
-        <Main data={data}/>
+
+        <Modal 
+          show={this.state.show} 
+          onHide={this.handleClose}
```

Comment:
- Once you move the Modal code into the newly created `SelectedBeast`, these props will be used. Double check what you are passing here to the `show` prop 👍